plugins/redis/redis.cpp

```diff
@@ -247,6 +248,16 @@ public :
             connection->unsubscribe();
     }
     inline Connection * operator -> () const { return connection.get(); }
+    inline void setown(Connection * _connection) { connection.setown(_connection); }
+    void handleException(IException * error)
+    {
+        if (connection)
+            connection->freeContext();
+
+        StringBuffer msg;
+        error->errorMessage(msg);
+        throw makeStringException(error->errorCode(), msg);
```

Comment:
This will leak error - but why not just throw the original exception (i.e. replace these three lines with

throw error;